### PR TITLE
Feat: need support cron hpa

### PR DESCRIPTION
**Pull request by StevenLeiZhang**
Signed-off-by: StevenLeiZhang <zhangleiic@163.com>

fix:#573

### Description of your changes

build one addon based on https://github.com/AliyunContainerService/kubernetes-cronhpa-controller

After addon installed, one new Trait is exposed, which is named "cronhpa". User could use it to set cronhpa for Component.

### How has this code been tested?
test it locally.

1. install it
```
cd experimental/addons/cronhpa
vale addon enable ./
```

2. wait for it is up
3.run test Application
```
cd examples/cronhpa
kubectl apply -f cronhpa-sample.yaml

```

4. check the pod amount of helloworld Application.
It will scale up and down every 30s.

### Checklist

I have:

- [ ] Title of the PR starts with type (e.g. `[Addon]` , `[example]` or `[Doc]`).
- [ ] Updated/Added any relevant [documentation](https://kubevela.io/docs/reference/addons/overview) and [examples](https://github.com/kubevela/catalog/tree/master/examples).
- [ ] New addon should be put in [experimental](https://github.com/kubevela/catalog/tree/master/experimental/addons).
- [ ] Update addon should modify the `version` in `metadata.yaml` to generate a new version.

####  Verified Addon promotion rules

If this pr wants to promote an experimental addon to verified, you must check whether meet these conditions too:
  - [ ] This addon must be tested by addon's [e2e-test](./test/e2e-test/addon-test) to guarantee this addon can be enabled successfully.
  - This addon must have some basic but necessary information.
    - [ ] An accessible icon url and source url defined in addon's `metadata.yaml`.
    - [ ] A detail introduction include a basic example about how to use and what's the benefit of this addon in `README.md`.
    - [ ] Also provide an introduction in KubeVela [documentation](https://kubevela.net/docs/reference/addons/overview).
    - [ ] It's more likely to be accepted if useful examples provided in example [dir](examples/).

### CODE REVIEW

**Review by wangyikewxgm — COMMENTED** [review]
Cool works！！

File: experimental/addons/cronhpa/template.cue [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+			{
+				type: "k8s-objects"
+				name: "cronhpa-CRD"
+				properties: objects: [
+					cronhpaCRD,
+				]
+			},
+			cronController]
+		policies: [{
+			type: "topology"
```
Comment (wangyikewxgm) [inline experimental/addons/cronhpa/template.cue:24]: We can add a keep resource GC policy for CRD resource. Thus, disabling this  addon wouldn't affect the application. Please refer to : https://github.com/kubevela/catalog/blob/374ce88aca03e7963a14bbd060a83547ae9330fc/addons/fluxcd/template.cue#L90

Comment (StevenLeiZhang) [reply experimental/addons/cronhpa/template.cue:24]: > We can add a keep resource GC policy for CRD resource. Thus, disabling this addon wouldn't affect the application. Please refer to :
> 
> https://github.com/kubevela/catalog/blob/374ce88aca03e7963a14bbd060a83547ae9330fc/addons/fluxcd/template.cue#L90

updated

**Review by wangyikewxgm — DISMISSED** [review]
LGTM

**Review by wonderflow — COMMENTED** [review]
Please add applyonce policy in example

**Comment by StevenLeiZhang** [comment]
> Please add applyonce policy in example

Done

**Review by wonderflow — APPROVED** [review]

**Pull request merged**